### PR TITLE
TASK-5809 - Implement new ADMIN UI - Organization section

### PR DESCRIPTION
**Pull request by gpveronica**
This PR enables users opencga / organization admins to admin key Organizations features through the Admin interface.

### CODE REVIEW

File: src/webcomponents/organization/admin/group-admin-grid.js [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+        }
+        super.update(changedProperties);
+    }
+
+    updated(changedProperties) {
+        if (changedProperties.size > 0 && this.active) {
+            if (this.groups?.length > 0) {
+                this.renderLocalTable();
+            }
+            this.requestUpdate();
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/group-admin-grid.js]: Please remove `this.requestUpdate()` from there, as it is performing unnecessary renders.

File: src/webcomponents/commons/layouts/custom-vertical-navbar.js [hunk 2]
```diff
@@ -301,7 +306,11 @@ export default class CustomVerticalNavBar extends LitElement {
                                     <hr class="mt-0 mb-3">
                                     <!-- TODO: CONTENT in a div -->
                                     <div class="settings-content-wrapper" id="settings-content-wrapper">
-                                        ${subItem.render(this.opencgaSession, this.study)}
+                                        ${
+                                        (this.organization) ?
+                                            subItem.render(this.opencgaSession, this.organization) :
+                                            subItem.render(this.opencgaSession, this.study)
+                                        }
```
Comment (jmjuanes) [inline src/webcomponents/commons/layouts/custom-vertical-navbar.js:313]: Why is this needed? The organization is already in `this.opencgaSession.organization`.

File: src/webcomponents/organization/admin/group-admin-permissions-update.js [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+                visible: () => {
+                    return UtilsNew.isNotEmpty(this.updatedFields);
+                },
+                notificationType: "warning",
+            },
+        };
+    }
+
+    #initPermissions() {
+        debugger
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/group-admin-permissions-update.js]: Please remove **ALL** the `debugger` lines of this file

File: src/webcomponents/organization/admin/organization-admin-audit.js [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+                    ${~this._config.filter.sections[0].filters.findIndex(field => field.id === "status") ? html`
+                        <!-- Status -->
+                        <div class="col-12">
+                            <select-field-filter2
+                                .data="${this.statusTypeValues}"
+                                .value=${this.query?.status}
+                                .config=${{
+            placeholder: "Status: All",
+            liveSearch: false,
+        }}
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/organization-admin-audit.js:326]: 1. **URGENT** The component `select-field-filter2` is deprecated and have been removed. Please switch to `select-field-filter`.
2. Please fix the indentation of all the `.config` properties.
3. Why you need the entire `this._config` for the `.config` properties of the first and second selects? (lines 274 and 291).

File: src/webcomponents/organization/admin/organization-admin-add.js
Comment (jmjuanes) [inline src/webcomponents/organization/admin/organization-admin-add.js:1]: Please remove this empty file.

File: src/webcomponents/organization/admin/organization-admin-owner-update.js
Comment (jmjuanes) [inline src/webcomponents/organization/admin/organization-admin-owner-update.js:1]: Please remove this empty file.

File: src/webcomponents/user/user-password-change.js [hunk 5]
```diff
@@ -16,55 +17,101 @@ export default class UserPasswordChange extends LitElement {
...
+        this._config = this.getDefaultConfig();
+        this.requestUpdate();
+    }
+
+    update(changedProperties) {
+        if (changedProperties.has("displayConfig")) {
+            this.displayConfig = {
+                ...this.displayConfigDefault,
+                ...this.displayConfig
+            };
```
Comment (jmjuanes) [inline src/webcomponents/user/user-password-change.js:63]: I would recommend not to change the original `this.displayConfig` object. Instead, merge both objects in the `getDefaultConfig` method:

```
getDefaultConfig() {
    return {
        displayConfig: {
            ...this.displayConfigDefault,
            ...this.displayConfig,
        },
        // ...
    };
}
```

File: src/webcomponents/study/admin/study-create.js [hunk 6]
```diff
@@ -118,22 +118,22 @@ export default class StudyCreate extends LitElement {
...
+            .then(() => {
                 this.study = {};
                 this._config = this.getDefaultConfig();
                 NotificationUtils.dispatch(this, NotificationUtils.NOTIFY_SUCCESS, {
                     title: "Study Create",
                     message: "New study created correctly"
                 });
-                LitUtils.dispatchCustomEvent(this, "sessionUpdateRequest");
+                LitUtils.dispatchCustomEvent(this, "sessionUpdateRequest", {}, {}, error);
+                LitUtils.dispatchCustomEvent(this, "studyCreate", {}, {}, error);
```
Comment (jmjuanes) [inline src/webcomponents/study/admin/study-create.js]: At this point, `error` is not defined yet (because it is assigned in the `catch`), why do you need to pass it to both events dispatchers?

File: src/webcomponents/project/project-update.js [hunk 7]
```diff
@@ -51,6 +51,7 @@ export default class ProjectUpdate extends LitElement {
     }
 
     #init() {
+        debugger
```
Comment (jmjuanes) [inline src/webcomponents/project/project-update.js]: Please remove the `debugger` lines on this file.

File: src/webcomponents/organization/admin/user-admin-update.js [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+        if (this.userId && this.opencgaSession) {
+            const params = {
+                organization: this.organization.id,
+            };
+            let error;
+            this.#setLoading(true);
+            this.opencgaSession.opencgaClient.users()
+                .info(this.userId, params)
+                .then(response => {
+                    this.user = UtilsNew.objectClone(response.responses[0].results[0]);
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/user-admin-update.js:135]: Do you really need to generate a clone of the result returned by OpenCGA?

File: src/webcomponents/organization/admin/user-admin-password-reset.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                this.onSubmit();
+            },
+            cancel: () => {
+                this._user = null;
+                LitUtils.dispatchCustomEvent(this, "closeNotification", null);
+            },
+        });
+    }
+
+    getDefaultConfig() {}
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/user-admin-password-reset.js]: Please fix the `getDefaultConfig` method, as it is not returning anything:

```
getDefaultConfig() {
    return {};
}
```

File: src/webcomponents/organization/admin/user-admin-password-reset.js
Comment (jmjuanes) [inline src/webcomponents/organization/admin/user-admin-password-reset.js:1]: This component is used?

File: src/webcomponents/organization/admin/user-admin-password-reset.js [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+            `,
+            ok: () => {
+                this.onSubmit();
+            },
+            cancel: () => {
+                this._user = null;
+                LitUtils.dispatchCustomEvent(this, "closeNotification", null);
+            },
+        });
+    }
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/user-admin-password-reset.js]: A `render` method without a `return`?????

File: src/webcomponents/organization/admin/user-admin-password-change.js [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+    update(changedProperties) {
+        if (changedProperties.has("userId")) {
+            this.userIdObserver();
+        }
+        if (changedProperties.has("displayConfig")) {
+            this._config = this.getDefaultConfig();
+            if (!this._config?.notification) {
+                this.#initConfigNotification();
+            }
+        }
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/user-admin-password-change.js:95]: The `displayConfig` is not used, why do you need it?

File: src/webcomponents/organization/admin/user-admin-create.js [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+            defaultLayout: "horizontal",
+            buttonOkText: "Create"
+        };
+        this._config = this.getDefaultConfig();
+    }
+
+    #initOriginalObjects() {
+        this._user = {};
+        this._config = this.getDefaultConfig();
+        this.requestUpdate();
```
Comment (jmjuanes) [inline src/webcomponents/organization/admin/user-admin-create.js]: Please do not call `requestUpdate` here.

**Review by jmjuanes — DISMISSED** [review]

File: src/webcomponents/organization/admin/organization-admin-audit.js (hunk 4 above)
Comment (gpveronica) [reply src/webcomponents/organization/admin/organization-admin-audit.js:326]: This code is currently in progress (and commented out its use) and was intended to be completed once the fix in OpenCGA for running audits for organizations was developed. I will address the issues mentioned in this comment in the next related task when the necessary fix in the back is ready.

File: src/webcomponents/user/user-password-change.js (hunk 5 above)
Comment (gpveronica) [reply src/webcomponents/user/user-password-change.js:63]: I completely agree. However, we'll need to discuss this further and ensure the changes are propagated across IVA for consistency. I coded it this way because it follows the agreed-upon template used in all views and creates. I'll leave it as it is for now until we reach an agreement.

File: src/webcomponents/study/admin/study-create.js (hunk 6 above)
Comment (gpveronica) [reply src/webcomponents/study/admin/study-create.js]: My bad on the cut&paste...

File: src/webcomponents/organization/admin/user-admin-update.js (hunk 8 above)
Comment (gpveronica) [reply src/webcomponents/organization/admin/user-admin-update.js:135]: Not really, just followed the code in opencga-update.js. I'll make the necessary changes in the next task.

File: src/webcomponents/commons/layouts/custom-vertical-navbar.js (hunk 2 above)
Comment (gpveronica) [reply src/webcomponents/commons/layouts/custom-vertical-navbar.js:313]: This code was developed some time before the organization object was included in this.opencgaSession. It will be refactored accordingly in the next task

File: src/webcomponents/organization/admin/user-admin-password-reset.js
Comment (gpveronica) [reply src/webcomponents/organization/admin/user-admin-password-reset.js:1]: Yes

File: src/webcomponents/organization/admin/user-admin-password-change.js (hunk 11 above)
Comment (gpveronica) [reply src/webcomponents/organization/admin/user-admin-password-change.js:95]: This component is not in use yet and its use has been commented out. I will work on it if we need it.

**Review by j-coll — APPROVED** [review]
